src/call_context.rs

```diff
@@ -295,7 +296,10 @@ impl<'a> CallContext<'a> {
     }
 
     /// Reconcile the gas usage across the stack.
-    fn gas_reconciliation(&mut self) -> Result<(), VMError> {
+    fn gas_reconciliation(
+        &mut self,
+        gas_meter: &mut GasMeter,
+    ) -> Result<(), VMError> {
```

Comment:
We pass `gas_meter` as argument only because we want to set it, we actually never do anything with it. I believe it should be a better API at this point having something like:
```rust
    /// Returns a new `GasMeter`with the reconciled gas usage across the stack.
    fn gas_reconciliation(
        &mut self,
    ) -> Result<GasMeter, VMError> {
        // If there is more than one [`StackFrame`] on the stack, then the
        // gas needs to be reconciled.
        if self.stack.len() > 1 {
            let len = self.stack.len() - 2;
            let spent = self.gas_meter().spent();
            let parent_meter = &mut self.stack[len].gas_meter;

            parent_meter.charge(spent)?
        }
        Ok(self.gas_meter().clone())
    }
```

And therefore when we need to call it:
```rust
*gas_meter = self.gas_reconciliation()?;
```
It also a more functional approach, in line with this PR.

Reply:
Implemented as suggested